src/query_builder.ts

```diff
@@ -22,7 +27,7 @@ export default function interpolateKustoQuery(query: string, scopedVars?: Scoped
     return values?.value ?? match;
   });
 
-  return query.replace(/\$__escapeMulti\(('[^]*')\)/gi, (match, p1) => escape(p1));
+  return query.replace(/\$__escapeMulti\(([^\|]*[^\)])\)/gi, (match, p1) => escape(replace(p1)));
```

Comment:
same, why did you need to change the regex?

Reply:
The original regex does not work here as values are no longer quoted because we aren't doing the replace and interpolation before executing this function. However, I think the original macro regexp is the better one now: `\$__([_a-zA-Z0-9]+)\(([^\)]*)\)` wdyt? This could be used above also.

Reply:
if that regexp works for this case yeah, better to call this `escape` in the first regex evaluation, we can skip this special case